src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -277,6 +277,14 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
   _handleKeydown(event: KeyboardEvent): void {
     const keyCode = event.keyCode;
 
+    // Prevent the default action on all escape key presses. This is here primarily to bring IE
+    // in line with other browsers. By default, pressing escape on IE will cause it to revert
+    // the input value to the one that it had on focus, however it won't dispatch any events
+    // which means that the model value will be out of sync with the view.
+    if (keyCode === ESCAPE) {
```

Comment:
Would it make sense to specifically target IE with this?

Reply:
I suppose, but other browsers don't do anything with it anyway so it wasn't worth changing the constructor signature.